core/templates/services/playthrough.service.spec.ts

```diff
@@ -298,6 +325,99 @@ describe('PlaythroughService', () => {
       });
     });
 
+    it('should identify disjoint cycle rotations', () => {
+      spyOn(stopwatchObjectFactory, 'create').and.returnValue(
+        jasmine.createSpyObj('Stopwatch', {getTimeInSecs: 370, reset: null}));
+
+      playthroughService.recordExplorationStartAction('stateName1');
+      // 1 -> 2 -> 3 -> 1
```

Comment:
These annotations don't quite seem to match up to the blocks of code that follow them (e.g. this has 4 numbers and there are 4 lines, but then there are 3 numbers and 2 lines, followed by 4 numbers and 6 lines, so I'm confused).

Reply:
OK, I've replaced these with helper-functions.